### PR TITLE
Bump to 1.39

### PR DESCRIPTION
**Pull request by Mark-Simulacrum**
r? @Centril

### CODE REVIEW

File: src/bootstrap/bin/rustc.rs [hunk 1]
```diff
@@ -132,10 +132,8 @@ fn main() {
         cmd.arg("-Dwarnings");
         cmd.arg("-Drust_2018_idioms");
         cmd.arg("-Dunused_lifetimes");
-        // cfg(not(bootstrap)): Remove this during the next stage 0 compiler update.
-        // `-Drustc::internal` is a new feature and `rustc_version` mis-reports the `stage`.
-        let cfg_not_bootstrap = stage != "0" && crate_name != Some("rustc_version");
-        if cfg_not_bootstrap && use_internal_lints(crate_name) {
+        // `rustc_version` mis-reports the `stage`.
+        if crate_name != Some("rustc_version") && use_internal_lints(crate_name) {
```
Comment (petrochenkov) [inline src/bootstrap/bin/rustc.rs]: The "rustc_version" part was also supposed to be removed during the bump.
("Mis-reports the `stage`" means reporting "not bootstrap" while it was actually "bootstrap".)

**Review by Centril — APPROVED** [review]
`cfg(bootstrap)` related changes look good aside from @petrochenkov's note.

r=me when-green with that fixed

**Comment by Centril** [comment]
Seems like hashbrown is using `mem::uninitialized` making the bootstrap bump here fail in https://github.com/rust-lang/rust/pull/63541#issuecomment-521074027... @bors r-

cc @Amanieu

**Comment by Amanieu** [comment]
@Centril The errors in the log don't seem to have anything to do with hashbrown?

**Comment by Mark-Simulacrum** [comment]
Yeah, I think that's just cloudabi. Fixed (I think).

@bors r=Centril

**Pull request merged**